internal/shim-sev/src/start.S

```diff
@@ -25,6 +25,52 @@
 # arg1 %rdi  = address of SYSCALL_PAGE (boot_info)
 # arg2 %rsi  = shim load offset
 _start:
+.intel_syntax noprefix
+CheckSevFeature:
+    // Check if we have a valid (0x8000_001F) CPUID leaf
+    mov       eax, 0x80000000
+    cpuid
+
+    // This check should fail on Intel or Non SEV AMD CPUs. In future if
+    // Intel CPUs supports this CPUID leaf then we are guranteed to have exact
+    // same bit definition.
```

Comment:
I'm skeptical about this claim.

Reply:
Ooops, shamelessly copied.

Author: Brijesh Singh <brijesh.singh@amd.com>

https://github.com/AMDESE/ovmf/commit/e60af8a1ebb15bfcbf2ecc4afb6cf35084c847aa

Reply:
want me to remove the comments?